File: 0070-climbing-stairs/0070-climbing-stairs.java
public class Solution {

    public int climbStairs(int n) {
        if (n == 1) return 1; // Base case: Only one way to climb 1 step.

        // Initialize an array to store ways to climb each step.
        int[] dp = new int[n + 1];
        dp[1] = 1; // One way to climb 1 step.
        dp[2] = 2; // Two ways to climb 2 steps (1+1 or 2).

        // Fill the dp array for the remaining steps.
        for (int i = 3; i <= n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2]; // Sum of ways to climb (i-1) and (i-2) steps.
        }

        return dp[n]; // The number of ways to climb 'n' steps.
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 1:
            return 1  # Base case: Only one way to climb 1 step
        
        # Initialize variables to store the previous two values
        prev2 = 1  # Ways to climb 1 step
        prev1 = 2  # Ways to climb 2 steps
        
        # Calculate the ways for each step from 3 to n
        for i in range(3, n + 1):
            current = prev1 + prev2  # Current step ways
            prev2 = prev1  # Update prev2
            prev1 = current  # Update prev1
        
        return prev1 if n > 1 else prev2

# Example usage:
solution = Solution()
print(solution.climbStairs(2))  # Output: 2
print(solution.climbStairs(3))  # Output: 3
print(solution.climbStairs(5))  # Output: 8
